_drafts/code-properties-and-language-features.md

```diff
@@ -0,0 +1,50 @@
+I feel like I only just started learning good object-oriented design, and the world has moved on to functional programming! As a result, the OO principles that are fresh on my mind are exactly what people are moving away from. For example, as I've been checking out Elixir and Redux, I've had questions about whether a lack of certain OO features causes problems. For example, encapsulation allows you to isolate implementation knowledge to one part of your app, so that other parts of your app can use that functionality without being coupled to its implementation. Is it a problem that FP doesn't offer the same kind of encapsulation?
+
+It's difficult for me to find answers to these questions. As Yehuda Katz [tweeted][yehuda-1] [recently][yehuda-2], there's a tendency to describe programming advancements as radically new paradigms, and past approaches as terrible. There's no grappling with the arguments older approaches made.
+
+This leaves me with a conundrum: I bought into arguments about encapsulation increasing maintainability. Were those arguments true, or were they just something we made up to tell ourselves so we could build software in a way we liked?
```

Comment:
"change firewalls" is my own wording. The article has lines like:

> We propose instead that one begins with a list of difficult design decisions or design decisions which are likely to change. Each module is then designed to hide such a decision from the others.

and

> The second decomposition was made using "information hiding" [4] as a criterion. The modules no longer correspond to steps in the processing. … Every module in the second decomposition is characterized by its knowledge of a design decision which it hides from all others. Its interface or definition was chosen to reveal as little as possible about its inner workings.

The section about "Efficiency and Implementation" makes clear that they were probably thinking in terms of writing code using a macro-assembler (!) rather than even any of the particular languages we'd have in mind today.